src/main/java/org/svydovets/dao/GenericJdbcDAO.java

```diff
@@ -187,18 +198,118 @@ private PreparedStatement prepareSelectStatement(EntityKey<?> entityKey, Connect
         }
     }
 
-    private <T> T createEntityFromResultSet(EntityKey<T> entityKey, ResultSet resultSet) {
+    private <T> T createEntityFromResultSet(Class<T> clazz, ResultSet resultSet) {
         try {
-            T entity = entityKey.entityType().getConstructor().newInstance();
-            ResultSetParser.parseForEntity(entity, resultSet);
+            T entity = clazz.getConstructor().newInstance();
+            try {
+                for (Field field : entity.getClass().getDeclaredFields()) {
+                    field.setAccessible(true);
+                    if (isEntityField(field)) {
+                        var joinClazz = field.getType();
+                        var joinColumnName = ParameterNameResolver.resolveColumnName(field);

```

Comment:
I guess it must be "resolveJoinColumnName()" method instead of "resolveColumnName()"

Reply:
Do we have a test for this case?